corehq/apps/cleanup/management/commands/remove_deleted_cases_from_es.py

```diff
@@ -0,0 +1,52 @@
+from django.core.management.base import BaseCommand
+
+from corehq.apps.es import CaseES, CaseSearchES, case_adapter
+from corehq.form_processor.models.cases import CommCareCase, RebuildWithReason
+from corehq.form_processor.backends.sql.processor import FormProcessorSQL
+
+
+class Command(BaseCommand):
+    help = ('Remove cases from case index in elasticsearch if deleted')
+
+    def add_arguments(self, parser):
+        parser.add_argument('domain')
+        parser.add_argument('--dry-run', action="store_true")
+
+    def handle(self, domain, **options):
+        dry_run = options["dry_run"]
+        missing_case_ids = get_case_ids_missing_from_case_search(domain)
+        for case_id in missing_case_ids:
+            should_delete = should_case_be_deleted(case_id, domain)
+            if should_delete:
+                print(f"removing case {case_id} in domain {domain} from ES")
+                if not dry_run:
+                    case_adapter.delete(case_id, refresh=True)
+            else:
+                print(f"case {case_id} in domain {domain} is not deleted")
+
+
+def should_case_be_deleted(case_id, domain):
+    """
+    This rebuilds a case from its related forms to determine what state the case
+    should be in
+    :param case_id: case_id for CommCareCase
+    :param domain: domain name
+    :return: returns True if it is in the deleted state, and False otherwise
+    """
+    detail = RebuildWithReason(reason="check final state of case")
+    sql_case, lock_obj = FormProcessorSQL.get_case_with_lock(case_id, lock=False)
```

Comment:
So I'm totally a fan of this, and made the change in 68157dd74712ccd18b536b77304b524c15fef43a but have received feedback against this in the past due to our use of `_` for translating text. In this case, I'm not translating anything in this file so I think it is fine, but just flagging in case you are thinking of using this in the future 👍🏻